addons/point_of_sale/models/pos_session.py

```diff
@@ -277,41 +287,389 @@ def action_pos_session_closing_control(self):
+
+        credit_lines = [generate_line_args('_credit_amount', account, **amounts) for account, amounts in credit_amounts.items()]
+        debit_lines = [generate_line_args('_debit_amount', account, **amounts) for account, amounts in debit_amounts.items()]
+        return credit_lines + debit_lines
+
+    @api.model
+    def _credit_amount(self, partial_move_line_args, amount, amount_currency=None, currency=None):
+        """ complete the `partial_move_line_args` by adding 'credit' and 'debit' fields with
+            abs(`amount`) assign to the correct field.
+
+            TODO jcb: The following is a note about the required parameter of currency._convert method
+            required parameters: from_amount, to_currency, company, date
+        """
+        if amount > 0:
+            init_args = dict(credit=amount, debit=0.0, **partial_move_line_args)
+        else:
+            init_args = dict(credit=0.0, debit=abs(amount), **partial_move_line_args)
+        # negative amount_currency if credit
+        return dict(amount_currency=-amount_currency if amount > 0 else amount_currency,
+                    currency_id=currency.id,
+                    **init_args) if amount_currency else init_args
+
+    @api.model
+    def _debit_amount(self, partial_move_line_args, amount, amount_currency=None, currency=None):
+        """ complete the `partial_move_line_args` by adding 'credit' and 'debit' fields with
+            abs(`amount`) assign to the correct field.
+        """
+        if amount > 0:
+            init_args = dict(credit=0.0, debit=amount, **partial_move_line_args)
+        else:
+            init_args = dict(credit=abs(amount), debit=0.0, **partial_move_line_args)
+        # negative amount_currency if credit
+        return dict(amount_currency=amount_currency if amount > 0 else -amount_currency,
+                    currency_id=currency.id,
+                    **init_args) if amount_currency else init_args
+
+    @api.multi
+    def show_journal_entries(self):
+        # get all the linked moves to this move
+        move = self.move_id
+        lines = sum(move.line_ids\
+                    .filtered(lambda aml: aml.account_id.reconcile)\
+                    .mapped(lambda aml: [r.debit_move_id for r in aml.matched_debit_ids] if aml.credit > 0 else [r.credit_move_id for r in aml.matched_credit_ids]),
+                    [])
+        ids = [line.move_id.id for line in lines] + [move.id]
+        domain = [('id', 'in', ids)]
+
+        # call the account move action tree view with default filter (journal)
+        # and domain containing the calculated ids above
+        [action] = self.env.ref('account.action_move_journal_line').read()
+        action['domain'] = [('id', 'in', ids)]
+        action['context'] = json.dumps({'search_default_journal': 1})
+        return action
+
+    @api.multi
+    def show_payments_list(self):
+        [action] = self.env.ref('point_of_sale.action_show_pos_payment_button').read()
```

Comment:
as the action is only used here, and not in any menus, you can return it in python without xml reference,like:

eturn {
                'name': _('Pos Payments'),
                'type': 'ir.actions.act_window',
                'res_model': pos.payment',
                  ...............
            }